app/lib/actions/katello/content_view/publish.rb

```diff
@@ -70,35 +71,9 @@ def humanized_name
         end
 
         def run
-          view = ::Katello::ContentView.find(input[:content_view_id])
-          version = ::Katello::ContentViewVersion.find(input[:content_view_version_id])
-          output[:content_view_id] = view.id
-          output[:content_view_version_id] = version.id
-          unless view.composite?
-            output[:composite_version_auto_published] = []
-            output[:composite_view_publish_failed] = []
-            output[:composite_auto_publish_task_id] = []
-
-            # Iterate through the list of composites
-            # this component belongs to
-            view.component_composites.each do |cv_component|
-              if cv_component.latest? && cv_component.composite_content_view.auto_publish?
-                description = _("Auto Publish - Triggered by '%{component}'") %
-                                { :component => version.name }
-                begin
-                  task = ForemanTasks.async_task(::Actions::Katello::ContentView::Publish,
-                                          cv_component.composite_content_view,
-                                          description,
-                                          :triggered_by => version)
-                  output[:composite_auto_publish_task_id] << task.id
-                  output[:composite_version_auto_published] << task.input[:content_view_version_id]
-                rescue StandardError
-                  ::Katello::UINotifications::ContentView::AutoPublishFailure.deliver!(
-                                              cv_component.composite_content_view)
-                  output[:composite_view_publish_failed] << cv_component.composite_content_view.id
-                end
-              end
-            end
+          if input[:auto_publish]
+            event = ::Katello::EventQueue.push_event(::Katello::Events::PublishComponentComposites::EVENT_TYPE, input[:content_view_version_id])
+            output[:composite_auto_publish_event_id] = event.id
```

Comment:
We lost some insight into the tasks that get scheduled for auto publish. Not sure if putting the event_id here is useful at all since it'll probably be gone if it ever needs to be looked at. Though since auto-publish should be more reliable now the output here should be less valuable in general.

Reply:
yeah, its probably not useful at all, i'd not bother